### PR TITLE
0199 Binary Tree Right Side View solved using python with doctest

### PR DESCRIPTION
**Pull request by kondekarshubham123**
## Change Summary

Provide summary of changes with issue number if any.

Added my solution for 0199 Binary Tree Right Side View solved using python with proper doctest,
Please merge this PR under `hacktoberfest` label

Thank you,
Looking for more contribution ^-^ 

## Checklist

**Ideally all of them should be checked. If you haven't fulfilled the below requirements or even delete the entire checklist, your PR won't be reviewed.**

### General 

- [X] Make sure you've read [CONTRIBUTING.md](https://github.com/wingkwong/leetcode-the-hard-way/blob/main/CONTRIBUTING.md) 
- [X] Make sure you've followed the template format
- [X] Make sure you've started the app locally and verified all the content and all links (if applicable) must be accessible correctly
- [X] Make sure you've applied LaTex for variables, formulas and time / space complexity instead of using backticks
- [X] Make sure you've formatted your code well (Code Block is rendered using `<Tabs/>`)
      Below is a C++ example. You can include multiple langs using `<TabItem/>` if necessary
      
````
<Tabs>

<TabItem value="cpp" label="C++">
<SolutionAuthor name="@YOUR_ALIAS"/>

```cpp
// Your code goes here
// Prefer K&R Coding style
// See https://gist.github.com/jesseschalken/0f47a2b5a738ced9c845#why-kr for example
```

</TabItem>

</Tabs>
````
- [X] Make sure you've explained your code well (write comments above each line - not inline)
- [X] Make sure you've written your explanation well and it is easy to understand for beginners
- [X] Make sure you've included Time & Space Complexity Analysis
- [X] Make sure you've included a meaningful approach name for your solution. Don't leave it blank. e.g. `## Approach 1: Two Pointers`.

### For Tutorial Articles

- [X] Make sure you've explained your topic using 1 - 3 LC problems rather than using external problems. Solutions must be taken with explanations as well.
- [X] Make sure you've provided suggested problems at the end using the given format. See [here](https://raw.githubusercontent.com/wingkwong/leetcode-the-hard-way/main/tutorials/math/number-theory/binary-exponentiation.md) as an example. If the target solution is not available, leave `solutionLink` blank.
- [X] Make sure you've given credits / references if you use external resources. For an image, give credit under it. Otherwise, add a new section called References at the end (after Suggested Problems).

### CODE REVIEW

**Comment by wingkwong** [comment]
FYI. solution doesn't count in hacktoberfest.

**Comment by kondekarshubham123** [comment]
Okay, no problem consider it as contribution towards opensource.

File: solutions/0100-0199/0199-binary-tree-right-side-view-medium.md [hunk 1]
```diff
@@ -87,4 +87,76 @@ class Solution {
...
+    3       <-  3
+   / \
+  9   20    <-  20
+     /  \
+    15   7  <-  7
+
+Output: [3, 20, 7]
+"""
+from __future__ import annotations
+from typing import List, Optional
```
Comment (wingkwong) [inline solutions/0100-0199/0199-binary-tree-right-side-view-medium.md:110]: what is the reason to include these?

File: solutions/0100-0199/0199-binary-tree-right-side-view-medium.md [hunk 2]
```diff
@@ -87,4 +87,76 @@ class Solution {
...
+        >>> root.right = TreeNode(20)
+        >>> root.right.left = TreeNode(15)
+        >>> root.right.right = TreeNode(7)
+        >>> obj.rightSideView(root)
+        [3, 20, 7]
+        
+        >>> obj.rightSideView(None)
+        []
+
+        """
```
Comment (wingkwong) [inline solutions/0100-0199/0199-binary-tree-right-side-view-medium.md:138]: remove these lines

File: solutions/0100-0199/0199-binary-tree-right-side-view-medium.md [hunk 3]
```diff
@@ -87,4 +87,76 @@ class Solution {
...
+        right_view: list = []
+        if not root:
+            return right_view
+        dfs(root, 0, right_view)
+        return right_view
+
+# if __name__ == "__main__":
+#     import doctest
+
+#     doctest.testmod()
```
Comment (wingkwong) [inline solutions/0100-0199/0199-binary-tree-right-side-view-medium.md:158]: remove these lines

**Review by wingkwong — CHANGES_REQUESTED** [review]

File: solutions/0100-0199/0199-binary-tree-right-side-view-medium.md (hunk 1 above)
Comment (kondekarshubham123) [reply solutions/0100-0199/0199-binary-tree-right-side-view-medium.md:110]: Those imports are for annotations purpose only

File: solutions/0100-0199/0199-binary-tree-right-side-view-medium.md (hunk 2 above)
Comment (kondekarshubham123) [reply solutions/0100-0199/0199-binary-tree-right-side-view-medium.md]: removed these lines

File: solutions/0100-0199/0199-binary-tree-right-side-view-medium.md (hunk 3 above)
Comment (kondekarshubham123) [reply solutions/0100-0199/0199-binary-tree-right-side-view-medium.md]: removed these lines

**Comment by kondekarshubham123** [comment]
Hi @wingkwong , suggested changes have been implemented

File: solutions/0100-0199/0199-binary-tree-right-side-view-medium.md (hunk 1 above)
Comment (kondekarshubham123) [reply solutions/0100-0199/0199-binary-tree-right-side-view-medium.md:110]: removing these lines

**Review by wingkwong — APPROVED** [review]

**Pull request merged**